### PR TITLE
Add action

### LINKED ISSUE #2729: Github Action for downloading ISPC in GH workflows
Something that would be nice would be to have an official GH action for ISPC that one can use as part of the workflow when building code code. 

https://docs.github.com/en/actions/creating-actions/about-custom-actions

What this essentially means is that the action should download the appropriate compiler. It would likely be good to support both stable and latest compiler as that way more than one build could be issues (one stable / on latest) to catch regressions.

### PR DESCRIPTION
**Pull request by nurmukhametov**
This PR contains a reworked version of the action from PR #1. There are two main differences: the use of `actions/tool-cache` and deriving default values for `architecture`/`platform`.

@ScatteredRay, thank you for your contribution. Your work was incredibly valuable and served as a solid foundation for this PR. It provided the necessary insights and direction, making the reworking process much smoother and more efficient.

This PR also includes a `README.md` derived from the draft version proposed by @maedoc.

Additionally, `DEVELOPMENT.md` and `.gitignore` were added to simplify the development process.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Install Dependencies
```
Comment (aneshlya) [inline DEVELOPMENT.md:7]: Please include prerequisites (e.g., Node.js version) before the installation steps.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,71 @@
-# install-ispc-action
-Github Action to install ISPC compiler
+# Install ISPC GitHub Action
```
Comment (aneshlya) [inline README.md:3]: Include build status badges at the top of the README for better visibility.

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+import { downloadTool, extractTar, extractZip } from '@actions/tool-cache';
+import { addPath, setFailed, getInput } from '@actions/core';
+import { platform as osPlatform, arch as osArch } from 'node:os';
+
+
+async function getLatestVersion() {
+    // Note: The GitHub API is rate limited to 60 requests per hour for unauthenticated users.
+    // I am not quite sure that it is reliable to query the latest version from
+    // the GitHub API because of the rate limits.
+    let response = await fetch(`https://api.github.com/repos/ispc/ispc/releases/latest`);
```
Comment (aneshlya) [inline src/main.js]: I don't know Javascript but it seems that there is `let` and `const` keywords (both can be used to declare variables in JavaScript). 
`let`: Allows you to reassign the variable after its initial declaration.
`const`: Does not allow reassignment. Once a value is assigned, it cannot be changed.

In this script most of the variables are `const`.

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (aneshlya) [inline src/main.js:1]: The formatting of this file is strange (no space between `if` and `(`), space between `Error (``. Hard ro read. Could you use some JS beautifier? Like https://beautifier.io/

**Review by aneshlya — APPROVED** [review]
Looks good! I just have a couple of minor style comments.

File: src/main.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+            const repoPath = 'ispc-repo';
+            if (!existsSync(repoPath)) {
+                execSync(`git clone --depth=1 https://github.com/ispc/ispc.git ${repoPath}`);
+            }
+
+            console.log('Fetching latest tags from the repository...');
+            execSync(`git -C ${repoPath} fetch --tags`);
+
+            let latestTagSHA = execSync(`git -C ${repoPath} rev-list --tags --max-count=1`).toString().trim();
+            version = execSync(`git -C ${repoPath} describe --tags ${latestTagSHA}`).toString().trim();
```
Comment (aneshlya) [inline src/main.js]: Should we remove the repo after determining the version?

File: DEVELOPMENT.md (hunk 1 above)
Comment (nurmukhametov) [reply DEVELOPMENT.md:7]: I have included a sentence that Node.js required, although I am not quite sure that it is worth mentioning any version there. My setup uses v22.2.0 whereas, e.g., `ubuntu-latest` uses v18.* and it looks fine.

File: README.md (hunk 2 above)
Comment (nurmukhametov) [reply README.md:3]: Done

File: src/main.js (hunk 4 above)
Comment (nurmukhametov) [reply src/main.js:1]: I have added `run lint` and `run lint-fix` commands to invoke linter. Surprisingly enough, it also fixes `let/const` mess in variable declarations.

File: src/main.js (hunk 3 above)
Comment (nurmukhametov) [reply src/main.js]: It looks like the linter fixed it.

File: src/main.js (hunk 5 above)
Comment (nurmukhametov) [reply src/main.js]: Fixed

**Review by aneshlya — APPROVED** [review]
LGTM!

**Pull request merged**